app/models/message.rb

```diff
@@ -1,17 +1,33 @@
 class Message < ApplicationRecord
+  belongs_to :assistant
   belongs_to :conversation
-  delegate :assistant, to: :conversation
   belongs_to :content_document, class_name: "Document", optional: true
   belongs_to :run, optional: true
 
   has_many :documents, dependent: :destroy
 
   enum role: %w[user assistant].index_by(&:to_sym)
 
+  before_validation :set_default_role, on: :create
+  before_validation :create_conversation, on: :create, if: -> { conversation.blank? }
+
+  validates :content_text, :role, presence: true
   validates :run, presence: true, if: -> { assistant? }
-  validates :role, presence: true
 
-  after_create_commit -> {
-    broadcast_append_to conversation, partial: "messages/message", locals: {scroll_into_view: true, conversation: conversation}
-  }
+  after_create_commit :broadcast_message
+
+
+  private
+
+  def create_conversation
+    self.conversation = Conversation.create!(user: Current.user, assistant: assistant)
+  end
```

Comment:
Note: Current is still used even when creating Messages. I think this a good example of the intended use of Current. It might seem a bit odd to have Current being access from a model, but sometimes the Model creation does indeed depend on context.

I don't anticipate we will ever create Messages outside of a controller when a Conversation will need to be initiated like this then Current.user can be set (for example within a worker or a test). In fact, I do it within the message_test file.

Reply:
Won't we be creating messages from the worker when receiving from the backend api?

Reply:
Those messages will specify the conversation so we'll avoid the need for Current with them.